Source/NSWindow.m

```diff
@@ -2737,10 +2737,24 @@ - (void) applicationDidChangeScreenParameters: (NSNotification *)aNotif
   newFrame.origin.y += newScreenFrame.size.height - oldScreenFrame.size.height;
   // Screen X origin change. Screen width change shouldn't affect our frame.
   newFrame.origin.x += newScreenFrame.origin.x - oldScreenFrame.origin.x;
-  [self setFrame: newFrame display: NO];
-  if (_autosaveName != nil)
+
+  if (_windowNum)
     {
-      [self saveFrameUsingName: _autosaveName];
+      /* Call backend's `placewindow::` directly because our origin in OpenStep 
+         coordinates might left unchanged and `setFrame:display:` has check 
+         for it. */
+      [GSServerForWindow(self) placewindow: newFrame : _windowNum];
+      if (_autosaveName != nil)
```

Comment:
I think this part belongs after the outer if. Even when the window was not instantiated the frame could be saved.

As for the rest of the code I really would prefer not to duplicate this code. Maybe we could use a new helper method from both places?

Reply:
Done. I hope it looks much better.